src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2928,6 +2930,57 @@ void learnFromEqualsMethodArguments(BoundExpression left, TypeWithState leftType
             }
         }
 
+        private void LearnFromCompareExchangeMethod(MethodSymbol method, BoundCall node, ImmutableArray<VisitArgumentResult> results)
+        {
+            // easy out
+            var parameterCount = method.ParameterCount;
+            if (parameterCount != 3
+                || method.MethodKind != MethodKind.Ordinary
+                || !method.IsStatic
+                || method.Name != WellKnownMembers.GetDescriptor(WellKnownMember.System_Threading_Interlocked__CompareExchange).Name)
+            {
+                return;
+            }
```

Comment:
This seems like overkill. Consider removing completely, or simply check `Name`.

Reply:
I was wondering about this. Checking whether a method is a particular static method seems cheaper than all the checks that needed to happen for IEquatable/IEqualityComparer.Equals in my previous PR. How about I just delete it.

Reply:
Yeah I would delete this. If we think this stuff is worth it we could do it in the symbol comparison.